### PR TITLE
cadence: fix dbus,libjack and missing dependancies

### LINKED ISSUE #55574: cadence doesn't see jack
## Issue description
cc maintainer: @bignaux

Some of the tools included with cadence need jack.
For example catia.
When I run it, I get a dialog saying: ``JACK is not available in this system, cannot use this application.``

### Steps to reproduce
```
nix-env -iA nixos.cadence 
catia
```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.15-rt12, NixOS, 19.03pre168781.929cc78363e (Koi)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(bart): `""`
 - channels(root): `"nixos-19.03pre168781.929cc78363e"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### PR DESCRIPTION
**Pull request by bignaux**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

This should fix #55574

### CODE REVIEW

**Comment by bignaux** [comment]
@GrahamcOfBorg build cadence
@worldofpeace ping

File: pkgs/applications/audio/cadence/default.nix [hunk 1]
```diff
@@ -65,10 +83,11 @@
...
     rm -f ${script}
-    makeWrapper ${python3Packages.python.interpreter} ${script} \
-      --set PYTHONPATH "$PYTHONPATH:${outRef}/share/cadence" \
-      ''${qtWrapperArgs[@]} \
-      --add-flags "-O ${source}"
+    makeQtWrapper ${source} ${script} \
+      --prefix PATH : "${lib.makeBinPath [
+        jack_capture # cadence-render
+        pulseaudioFull # cadence, cadence-session-start
+        ]}"
```
Comment (worldofpeace) [inline pkgs/applications/audio/cadence/default.nix]: doesn't this remove all the important PYTHONPATH changes to the wrapper?
Can you also indent properly?

Comment (bignaux) [reply pkgs/applications/audio/cadence/default.nix]: python3 seems to be better than python3Packages, then no need for propagatedBuildInputs and such things in the wrapper. I test heavily to catch others bugs. BTW, i've another thing i'd fix, and i've no idea how to do that properly, I need /nix/store/v433wawv8hl23kb2c2qk0lgg86sainc1-a2jmidid-8/share/dbus-1/services/org.gna.home.a2jmidid.service to be activated .
we can complete with a :

  postInstall = ''
    # D-Bus activation
    install -Dm755 ${a2jmidid}/share/dbus-1/services/org.gna.home.a2jmidid.service -t $out/share/dbus-1/services/
  ''
but user need to reboot for the service to be activated by dbus. There is a lot of issue with dbus activation #17943 for example.

**Comment by magnetophon** [comment]
This indeed fixes #55574.

**Comment by zarybnicky** [comment]
I can confirm that cadence works. Trying out claudia, packaged here as well, I'm encountering

```
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.NameHasNoOwner: Could not get owner of name 'org.ladish': no such name
```

I found one mention of the problem in mageia - https://bugs.mageia.org/show_bug.cgi?id=17867 - where they've found out that packaging ladish is necessary for claudia to run.

**Comment by bignaux** [comment]
Yes i write a comment on that . anyway, package the death ladish is not impossible but i wont . https://github.com/NixOS/nixpkgs/pull/81098/files#diff-5e87607c20c10470e7a0346a3be0f6e2R16

**Comment by zarybnicky** [comment]
Aha, I missed that comment, sorry. In any case, a working cadence is already a nice step forward for NixOS Audio :)

**Comment by bignaux** [comment]
only if it is merged. I spend days to fix things in the repo, and get sometime frustrate to see some typo or no reaction. A good orga is measured by the ability to delegate.

File: pkgs/applications/audio/cadence/default.nix (hunk 1 above)
Comment (adisbladis) [reply pkgs/applications/audio/cadence/default.nix]: @worldofpeace The wrapped python interpreter contains hard-coded references to the correct paths. There is no need to set `PYTHONPATH` when using `withPackages`.

**Pull request merged**